catroid/src/main/java/org/catrobat/catroid/ui/ProjectUtils.kt

```diff
@@ -0,0 +1,130 @@
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+@file:JvmName("ProjectUtils")
+
+/**
+ * black listed bricks as defined in CATROID-681
+ *
+ * StartListeningBrick and WebRequestBrick or BackgroundRequestBrick or LookRequestBrick
+ * */
+
+package org.catrobat.catroid.ui
+
+import android.content.Context
+import android.content.Context.MODE_PRIVATE
+import androidx.appcompat.app.AlertDialog
+import org.catrobat.catroid.ProjectManager
+import org.catrobat.catroid.R
+import org.catrobat.catroid.content.Project
+import org.catrobat.catroid.content.bricks.BackgroundRequestBrick
+import org.catrobat.catroid.content.bricks.Brick
+import org.catrobat.catroid.content.bricks.ForeverBrick
+import org.catrobat.catroid.content.bricks.LookRequestBrick
+import org.catrobat.catroid.content.bricks.StartListeningBrick
+import org.catrobat.catroid.content.bricks.WebRequestBrick
+
+/**
+ * extension boolean function for List<Brick> data type.
+ * check if the list contains black-listed bricks
+ * */
+private fun List<Brick>.containsBlackListedBricks(): Boolean {
+    val startListeningBrickExists = any { brick ->
+        brick is StartListeningBrick
+    }
+    val backgroundRequestOrWebRequestBrickExists = any { brick ->
+        brick is WebRequestBrick || brick is BackgroundRequestBrick || brick is LookRequestBrick
+    }
+    return startListeningBrickExists and backgroundRequestOrWebRequestBrickExists
+}
+
+/**
+ * extension boolean function for Project data type.
+ * check if the project contains black-listed bricks
+ * */
+private fun Project.shouldDisplayWarning(): Boolean {
+    val brickList = arrayListOf<Brick>()
+    sceneList.forEach { scene ->
+        brickList.run {
+            scene.backgroundSprite.allBricks.forEach {
+                add(it)
+                if (it is ForeverBrick) {
+                    addAll(it.nestedBricks)
+                    add(it.allParts.last())
+                }
+            }
+            scene.spriteList.forEach { sprite ->
+                sprite.allBricks.forEach {
+                    add(it)
+                    if (it is ForeverBrick) {
+                        addAll(it.nestedBricks)
+                        add(it.allParts.last())
+                    }
+                }
+            }
+        }
+    }
+    return brickList.containsBlackListedBricks()
+}
+
+/**
+ * [only once]!
+ * show warning dialog for downloaded projects which contains black-listed bricks
+ * */
+
+fun showWarningDialogIfNeeded(context: Context) {
```

Comment:
```suggestion
fun showWarningForSuspiciousBricksOnce(context: Context) {
```
Ideally, no comments, instead use clearer names.

Reply:
true in the ideal world & non-open-source project.